### PR TITLE
Refactor PR oversee helpers into package

### PR DESCRIPTION
**Pull request by arthexis**
### Motivation

- Improve modularity and maintainability by splitting a large `pr_oversee.py` into focused modules for worktree handling, checks parsing, hygiene/test-plan heuristics, reply formatting, and orchestration.
- Reduce untyped `dict[str, Any]` usage by introducing typed payload fragments for PR/check/review payloads to improve discoverability and IDE support.
- Preserve existing runtime behavior and public import surface while enabling more granular unit testing and future policy adjustments.

### Description

- Create the `apps/repos/pr_oversee/` package and re-export the original public API from `apps.repos.pr_oversee` via `__init__.py` to preserve compatibility.
- Extracted patchwork/worktree helpers into `worktree.py` (pathing, reparse-point detection, venv linking, owned-path removal, and status parsing).
- Extracted check-rollup parsing, readiness gate logic, dependency detection/deduping into `checks.py` and moved hygiene/test-plan logic and review reply formatting into `hygiene.py` and `reply.py` respectively.
- Moved orchestration and the `PullRequestOverseer` command-runner integration into `service.py`, added `types.py` with TypedDict/dataclass payload structures, and updated tests to import the new modules directly.

### Testing

- Ran `python -m py_compile apps/repos/pr_oversee/*.py` and confirmed the new package modules compile successfully.
- Ran `pytest -q apps/repos/tests/test_pr_oversee.py`; after resolving a few missing-import issues during the refactor the suite completed with all tests passing (`44 passed, 2 warnings`).
- Ran project formatting (`black`) and repository hygiene checks (`git diff --check`); these completed without introducing test regressions.
- Note: the canonical local command `.venv/bin/python manage.py test run -- apps/repos/tests/test_pr_oversee.py` could not be executed in this environment because the repository virtualenv/bootstrap step required system-level Redis, so tests were validated with the available `pytest` execution instead.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a18636aa4608326a400f2d232fc817c)

## Refactor PR oversee helpers into package

This PR refactors the monolithic apps.repos.pr_oversee module into a focused package apps/repos/pr_oversee with a preserved public import surface, improving modularity, testability, and type safety while keeping runtime behavior unchanged.

### Changes

- New package and public re-exports:
  - Added apps/repos/pr_oversee/__init__.py that re-exports the package's public API via __all__ so existing imports continue to work.

- New modules:
  - types.py: Typed payload schemas (TypedDicts) for PR/check/review payloads, JSONValue alias, PullRequestOverseeError, CommandResult dataclass, and small coercion/parsing helpers. Reduces untyped dict[str, Any] usage.
  - checks.py: Check-rollup normalization and grouping (GOOD_CHECKS/BAD_CHECKS/PENDING_CHECKS, BAD_MERGE_STATES, VERSION_SUFFIX_SEPARATORS), check_rollup_state, readiness_gate (blockers/warnings computation with options to allow pending or require conversation resolution), dependency detection/normalization and deduplication helpers (is_dependency_pr, dependency_key, dependency_target_version, dependency_duplicates).
  - hygiene.py: File-change classification (README_RE), test-plan command generation (classify_changed_files, build_test_commands, changed_files_to_test_plan), and hygiene_report that inspects PR body and changed files for hygiene failures/warnings. (Note: changed_files_to_test_plan is defined twice in the file; the second definition overrides the first.)
  - reply.py: review_reply_summary to compose structured review-reply bodies (commit truncation, trimmed lists, conditional sections).
  - worktree.py: Patchwork worktree pathing and filesystem helpers (PATCHWORK_ENV_VAR, PATCHWORK_METADATA, PATCHWORK_OWNED_NOISE), default_patchwork_dir, patchwork_worktree_path, relative-path checks, Patchwork-owned noise detection, Windows reparse-point and venv linking helpers (_local_venv_link), and helper to detect missing git worktree errors.
  - service.py: Retains orchestration (PullRequestOverseer) and subprocess wrapper (CommandRunner) but delegates previous helper implementations to the new submodules and imports moved types/constants (notably PATCHWORK_METADATA and PATCHWORK_OWNED_NOISE from worktree, and types from types.py). Patchwork path helper previously present in service.py was relocated to worktree.py.

- Tests:
  - Updated apps/repos/tests/test_pr_oversee.py to import submodules directly and to assert package re-exports match submodule originals.
  - Adjusted a test for changed_files_to_test_plan command arguments.
  - Added dependency title parsing edge-case tests and a symlink-path handling test for _path_is_relative_to.
  - Test run: python -m py_compile succeeded for new modules; pytest -q apps/repos/tests/test_pr_oversee.py completed with 44 passed, 2 warnings.

### Notable details and behavior preservation

- Backward compatibility: __init__.py re-exports the original public API so external callers can continue using apps.repos.pr_oversee as before.
- Types: Introduction of TypedDicts and CommandResult makes payload handling explicit and reduces ad-hoc dict usage.
- Hygiene module caveat: changed_files_to_test_plan appears twice in hygiene.py; the latter definition is effective.
- No intended behavior changes: authors verified compilation and tests; manage.py-based canonical test runner could not be executed in this environment due to external Redis dependency, but pytest validation passed.
- Large refactor: service.py lost many inline helper implementations (moved to submodules) — reviewers should focus on import correctness, API surface in __init__.py, and that tests exercise the re-exported symbols.

### Benefits

- Clear separation of concerns (checks, hygiene, reply formatting, worktree, types, orchestration).
- Easier unit testing and granular policy adjustments.
- Stronger typing for payloads, improving maintainability.
- Preserves existing public API and runtime behavior via package re-exports.

[Review Change Stack](https://app.coderabbit.ai/change-stack/arthexis/arthexis/pull/8152?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: apps/repos/pr_oversee/worktree.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+    return root.expanduser() / f"{_slugify_path_segment(repo)}-pr-{number}"
+
+
+def _path_is_relative_to(path: Path, root: Path) -> bool:
+    try:
+        path.resolve().relative_to(root.resolve())
+    except (OSError, ValueError):
+        return False
+    return True
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/repos/pr_oversee/worktree.py]: Resolved after verification. Current worktree helper already uses Path.is_relative_to with an OSError guard, and the focused pr_oversee tests pass at commit 129c16252.

File: apps/repos/pr_oversee/hygiene.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+
+def build_test_commands(
+    apps: set[str], flags: Mapping[str, bool], generated_candidates: Iterable[str]
+) -> tuple[list[list[str]], list[str]]:
+    commands = [[sys.executable, MANAGE_PY, "check", "--fail-level", "ERROR"]]
+    notes: list[str] = []
+    test_paths = [f"apps/{app_label}/tests" for app_label in sorted(apps)]
+    if test_paths:
+        commands.append([sys.executable, MANAGE_PY, "test", *test_paths])
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/repos/pr_oversee/hygiene.py]: Addressed in commit 129c16252: generated test plans now use this repository's manage.py test run -- path command shape, and the test expectation was updated. Validation passed: py_compile for pr_oversee modules, manage.py check, manage.py test run -- apps/repos/tests/test_pr_oversee.py, and git diff --check.

File: apps/repos/tests/test_pr_oversee.py [hunk 3]
```diff
@@ -11,21 +11,43 @@
...
+    assert pr_oversee.CommandResult is CommandResult
+    assert pr_oversee.PullRequestOverseeError is PullRequestOverseeError
+    assert pr_oversee.PullRequestOverseer is PullRequestOverseer
+    assert pr_oversee.changed_files_to_test_plan is changed_files_to_test_plan
+    assert pr_oversee.dependency_duplicates is dependency_duplicates
+    assert pr_oversee.hygiene_report is hygiene_report
+    assert pr_oversee.readiness_gate is readiness_gate
+    assert pr_oversee.review_reply_summary is review_reply_summary
+    assert expected_exports <= set(pr_oversee.__all__)
+
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/repos/tests/test_pr_oversee.py]: Addressed in commit 129c16252: consolidated the duplicate package re-export tests, removed duplicate symlink safety coverage, and removed the duplicate hygiene declarations/dead changed_files_to_test_plan implementation. Validation passed: py_compile for pr_oversee modules, manage.py check, manage.py test run -- apps/repos/tests/test_pr_oversee.py, and git diff --check.
